api/v1beta2/types.go

```diff
@@ -232,6 +232,10 @@ type Instance struct {
 	// InstanceMetadataOptions is the metadata options for the EC2 instance.
 	// +optional
 	InstanceMetadataOptions *InstanceMetadataOptions `json:"instanceMetadataOptions,omitempty"`
+
+	// PrivateDNSNameOptions is the options for the instance hostname.
+	// +optional
+	PrivateDNSNameOptions *PrivateDNSNameOptions `json:"privateDnsNameOptions,omitempty"`
```

Comment:
```suggestion
	// PrivateDNSName allows to configure DNS options for the instance hostname.
	// +optional
	PrivateDNSName *PrivateDNSName `json:"privateDnsNameOptions,omitempty"`
```

Reply:
> Minor suggestion, totally optional, rest LGTM

This `PrivateDnsNameOptions` naming was taken directly from the AWS SDK `Instance` struct
```
// The options for the instance hostname.
PrivateDnsNameOptions *PrivateDnsNameOptionsResponse `locationName:"privateDnsNameOptions" type:"structure"`
```

But I'm completely ok with renaming it, if you think it will be more suitable :)

Reply:
Got it, yeah I saw although the `Options` or similar suffix usually are redundant in CRD's spec fields, given that they're already options to setup these values

Reply:
Same goes for the json name, would suggest to rename it